piksi_tools/console/update_view.py

```diff
@@ -509,30 +577,23 @@ def _compare_versions(self):
                                   actions=[prompt.close_button]
                                  )
 
-        fw_update_prompt.text = \
+        if self.update_dl.index[self.piksi_hw_rev].has_key('fw'):
+          fw_update_prompt.text = \
+            "New Piksi firmware available.\n\n" + \
+            "Please use the Firmware Update tab to update.\n\n" + \
+            "Newest Firmware Version :\n\t%s\n\n" % \
+                self.update_dl.index[self.piksi_hw_rev]['fw']['version']
+        else:
+          fw_update_prompt.text = \
             "New Piksi firmware available.\n\n" + \
             "Please use the Firmware Update tab to update.\n\n" + \
             "Newest STM Version :\n\t%s\n\n" % \
-                self.update_dl.index['piksi_v2.3.1']['stm_fw']['version'] + \
+                self.update_dl.index[self.piksi_hw_rev]['stm_fw']['version'] + \
             "Newest SwiftNAP Version :\n\t%s\n\n" % \
-                self.update_dl.index['piksi_v2.3.1']['nap_fw']['version']
+                self.update_dl.index[self.piksi_hw_rev]['nap_fw']['version']
 
         fw_update_prompt.run()
 
```

Comment:
On a slow or nonexistent internet connection it seems like downloading the file makes the console lag when you first start it up.  I think that this is the reason that this version info download was in its own thread.  Can we make it that way again?